src/lib.rs

```diff
@@ -2325,6 +2345,39 @@ impl Build {
         .map(|x| x.to_owned()))
     }
 
+    /// Some platforms have multiple, compatible, canonical prefixes. Look through
+    /// each possible prefix for a compiler that exists and return it. The prefixes
+    /// should be ordered from most-likely to least-likely.
+    fn find_working_gnu_prefix(&self, prefixes: &[&'static str]) -> Option<&'static str> {
+        let windows_host = self
+            .get_host()
+            .map(|host| host.contains("windows"))
+            .unwrap_or(false);
+        let suffix = if self.cpp { "-g++" } else { "-gcc" };
+        let extension = if windows_host { ".exe" } else { "" };
+
+        for prefix in prefixes {
+            let target_compiler = format!("{}{}{}", prefix, suffix, extension);
+
+            if env::var_os("PATH")
+                .as_ref()
+                .and_then(|path_entries| {
+                    env::split_paths(path_entries)
+                        .find(|path_entry| path_entry.join(&target_compiler).exists())
```

Comment:
`.find(..).is_some()` can be replaced with `.any(...)`

Reply:
Thanks, I also didn't know about this construct. However, as part of inverting the two loops, this ended up being unnecessary. And in fact the whole thing got collapsed down to a single statement.